src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseRepository.java

```diff
@@ -164,10 +164,10 @@ public void linkSealedClaimDocument(String authorisation, Claim claim, URI docum
     public void saveReDetermination(
         String authorisation,
         Claim claim,
-        ReDetermination reDetermination,
-        String submitterId
+        ReDetermination reDetermination
     ) {
-        throw new NotImplementedException("We do not implement CCD yet");
+        coreCaseDataService.saveReDetermination(authorisation, claim.getId(), reDetermination,
+            CaseEvent.valueOf("REFER_TO_JUDGE_BY_" + reDetermination.getPartyType().name()));
```

Comment:
Please change it to choosing Enum based or party type. so that we can keep partytype name and this event independant

Reply:
actioned